### PR TITLE
Add clang32/64 MSYS environments

### PR DESCRIPTION
**Pull request by eustas**
Update HWY to ToT.

Also re-enable AVX targets and CLink tests

### CODE REVIEW

**Comment by kmilos** [comment]
Please try the CLANG64 environment/toolchain instead of just switching compiler in the MINGW64 environment:

https://www.msys2.org/docs/environments/

https://github.com/msys2/setup-msys2/blob/main/examples/cmake.yml 

Also, instead of installing the entire "toolchain" metapackage w/ pacboy, to reduce CI time one can do only the "cc" metapackage (gives you the default compiler for the given msys environment) and other basic requirements.

**Comment by kmilos** [comment]
Same difference, you could be trying out CLANG32 then in addition to or vs MINGW32.

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -259,24 +268,24 @@ jobs:
...
-            mingw-w64-${{ matrix.arch }}-gtest
-            mingw-w64-${{ matrix.arch }}-libavif
-            mingw-w64-${{ matrix.arch }}-libjpeg-turbo
-            mingw-w64-${{ matrix.arch }}-libpng
-            mingw-w64-${{ matrix.arch }}-libwebp
-            mingw-w64-${{ matrix.arch }}-ninja
+          pacboy: >-
+            brotli:p
+            cmake:p
+            ${{ matrix.compiler }}:p
```
[the remark this answers is not in the document]
Comment (eustas) [reply .github/workflows/build_test.yml:274]: Cool! Thanks.

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -259,24 +264,23 @@ jobs:
...
+            cmake:p
+            gflags:p
+            giflib:p
+            gtest:p
+            libavif:p
+            libjpeg-turbo:p
+            libpng:p
+            libwebp:p
+            ninja:p
+            toolchain:p
```
Comment (kmilos) [inline .github/workflows/build_test.yml:279]: Nice, a much more intuitive action now 👍

Just out of curiousity, why was the entire `toolchain:p` needed? Usually just `cc:p` and (for C++ code) `gcc-libs:p` suffice, and might give some time savings by not installing all the toolchain packages... (Note that there is a `gcc-libs` metapackage available in clang environments for this reason, which will pull in libc++ and libunwind for you; there might missing corner cases though like openmp and libssp that are not part of that metapackage, do that's why I ask...)

Comment (eustas) [reply .github/workflows/build_test.yml:279]: `clang64` environment and with `cc` was missing libs. Will try with `gcc-libs` (well, package name is counterintuitive)

Comment (kmilos) [reply .github/workflows/build_test.yml:280]: Indeed, hence the lengthy explanation 😉 It was one way to extend the legacy mingw environments...

Comment (eustas) [reply .github/workflows/build_test.yml:279]: What's suspicious is that it tries to link 32-bit libs...

Comment (kmilos) [reply .github/workflows/build_test.yml:279]: The "32" is just the Win32 API, not the architecture...

Try adding `crt:p` to the list as well?

**Review by mo271 — DISMISSED** [review]

**Pull request closed without merging**